core/container.go

```diff
@@ -996,6 +998,70 @@ func (container *Container) Platform() (specs.Platform, error) {
 	return payload.Platform, nil
 }
 
+func (container *Container) Export(
+	ctx context.Context,
+	host *Host,
+	dest string,
+	bkClient *bkclient.Client,
+	solveOpts bkclient.SolveOpt,
+	solveCh chan<- *bkclient.SolveStatus,
+) error {
+	dest, err := host.NormalizeDest(dest)
+	if err != nil {
+		return err
+	}
+
+	payload, err := container.ID.decode()
+	if err != nil {
+		return err
+	}
+
+	st, err := payload.FSState()
+	if err != nil {
+		return err
+	}
+
+	stDef, err := st.Marshal(ctx, llb.Platform(payload.Platform))
+	if err != nil {
+		return err
+	}
+
+	cfgBytes, err := json.Marshal(specs.Image{
+		Architecture: payload.Platform.Architecture,
+		OS:           payload.Platform.OS,
+		OSVersion:    payload.Platform.OSVersion,
+		OSFeatures:   payload.Platform.OSFeatures,
+		Config:       payload.Config,
+	})
+	if err != nil {
+		return err
+	}
+
+	out, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+
+	defer out.Close()
+
+	return host.Export(ctx, bkclient.ExportEntry{
+		Type: bkclient.ExporterOCI,
```

Comment:
afaict this exporter type supports multiplatform too, so we should export it for each platform ([same as here](https://github.com/sipsma/dagger/blob/530446113eeeeffb2aa0388641ba30d9a1435158/core/container.go#L893-L893))

Reply:
I copied everything from `Publish` into a private function shared by `Export`, but one caveat is that the multi-platform OCI images created by it aren't compatible with `docker load` since they don't contain a `manifest.json`.

As a compromise I've made it include a `manifest.json` if there's only one platform, which should be compatible with the previous behavior.

Reply:
Great catch on that issue, thanks!